js/src/views/Dapps/dappsFetcher.js

```diff
@@ -0,0 +1,271 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import BigNumber from 'bignumber.js';
+import { pick, range, uniq } from 'lodash';
+import EventEmitter from 'eventemitter3';
+
+import Contracts from '~/contracts';
+import { hashToImageUrl } from '~/redux/util';
+import { bytesToHex } from '~/api/util/format';
+
+import builtinApps from './builtin.json';
+
+const BUILTIN_APPS_KEY = 'BUILTIN_APPS_KEY';
+let dappsFetcherInstance = null;
+
+export default class DappsFetcher extends EventEmitter {
```

Comment:
For now, probably keep it in dapps (like the fetcher was), longer term we can probably `s/redux/stores` and put it in there since it deals with all the global stores.